scipy/linalg/_decomp_lu.py

```diff
@@ -174,53 +182,166 @@ def lu(a, permute_l=False, overwrite_a=False, check_finite=True):
         Whether to check that the input matrix contains only finite numbers.
         Disabling may give a performance gain, but may result in problems
         (crashes, non-termination) if the inputs do contain infinities or NaNs.
+    p_indices : bool, optional
+        If ``True`` the permutation information is returned as row indices.
+        The default is ``False`` for backwards-compatibility reasons.
```

Comment:
can we name this more closely to `permute_l`? something like `permutation_indices` / `perm_indices` / `permutation_idx`...

In general, I'd be happier if the replace-Fortran-with-Cython effort was separate from such API changes. I'm particularly not a fan of return signatures that change based on arguments. This is not the first and only place we have such issues, but then it would be nice to have a clear view of what API we want to achieve, and migrate there once, with a clear:
```
if legacy:
    # raises DeprecationWarning
    return the_old_way
else:
    return the_new_way  # possibly packaged in some smarter way, like LUResult
```

It would IMO be a pity to hold up the unequivocally beneficial fortran clean-up by the discussion that's (IMO) necessary for such API refactors.

Reply:
There is a mailing list question about it that didn't receive much attention so I'm a bit undecided about the naming of `p_indices`. 

The return signature thing is very troubling me too but I don't know if we are ready to do it. It almost seems like a SciPy 2.0 type of clean up.

Reply:
Revived the mailing list about it